### PR TITLE
Add Approximately Equal protocol and Angle Tests

### PR DESCRIPTION
**Pull request by jbeenie**

### CODE REVIEW

File: PathKit/Sources/Angle.swift [hunk 1]
```diff
@@ -9,7 +9,7 @@ import Foundation
 
 public struct Angle: Equatable {
 
-    public static let π = Angle(radians: Double.pi)
+    public static let pi = Angle(radians: Double.pi)
```
Comment (alejandro-isaza) [inline PathKit/Sources/Angle.swift]: Not worth changing.

File: PathKit/Sources/Angle.swift [hunk 2]
```diff
@@ -27,6 +27,27 @@ public struct Angle: Equatable {
         }
     }
 
+    /// True if the angle is positive, false otherwise.
+    public var isPositive: Bool {
+        return 0.0 < radians
+    }
+
+    /// A positive normalized angle between 0 and 2 * .pi.
```
Comment (alejandro-isaza) [inline PathKit/Sources/Angle.swift]:
```suggestion
    /// A positive normalized angle between 0 and 2π.
```

File: PathKit/Sources/Angle.swift [hunk 3]
```diff
@@ -27,6 +27,27 @@ public struct Angle: Equatable {
...
+    public var positiveNormalized: Angle {
+        let twoPI = 2 * Angle.pi
```
Comment (alejandro-isaza) [inline PathKit/Sources/Angle.swift]: Let's define `public static let τ = Angle(Radians: 2 * Double.pi)`

File: PathKit/Sources/Angle.swift [hunk 4]
```diff
@@ -27,6 +27,27 @@ public struct Angle: Equatable {
...
+        let angle = Angle(radians: radians.truncatingRemainder(dividingBy: twoPI.radians))
+        return isPositive ? angle : angle + twoPI
+    }
+
+    /// Returns the angular distance between two angles as an angle ranging between 0 and 180 degrees.
```
Comment (alejandro-isaza) [inline PathKit/Sources/Angle.swift]:
```suggestion
    /// Returns the angular distance between two angles.
```

File: PathKit/Sources/Angle.swift [hunk 5]
```diff
@@ -27,6 +27,27 @@ public struct Angle: Equatable {
...
+    /// Eg: distance between 1 and -360 degrees is 1 degree.
```
Comment (alejandro-isaza) [inline PathKit/Sources/Angle.swift]: Let's format this comment better. Don't be afraid of some empty lines.

File: PathKit/Sources/Angle.swift [hunk 6]
```diff
@@ -27,6 +27,27 @@ public struct Angle: Equatable {
...
+    public static func distance(between lhs: Angle, rhs: Angle) -> Angle {
```
Comment (alejandro-isaza) [inline PathKit/Sources/Angle.swift]:
```suggestion
    public static func distance(_ lhs: Angle, _ rhs: Angle) -> Angle {
```

File: PathKit/Sources/Angle.swift [hunk 7]
```diff
@@ -27,6 +27,27 @@ public struct Angle: Equatable {
...
+        var angDiff = abs(lhs.positiveNormalized.radians - rhs.positiveNormalized.radians)
```
Comment (alejandro-isaza) [inline PathKit/Sources/Angle.swift]: No abbreviations please.
```suggestion
        var angleDifference = abs(lhs.positiveNormalized.radians - rhs.positiveNormalized.radians)
```

File: PathKit/Sources/Angle.swift [hunk 8]
```diff
@@ -103,3 +124,15 @@ public func sin(_ angle: Angle) -> Double {
...
 }
+
+extension Angle: Comparable {
+    public static func < (lhs: Angle, rhs: Angle) -> Bool {
+        return lhs.radians < rhs.radians
+    }
+}
+
+extension Angle: ApproximatelyEquatable {
+    public static func equal(lhs: Angle, rhs: Angle, accuracy: Angle) -> Bool {
```
Comment (alejandro-isaza) [inline PathKit/Sources/Angle.swift]:
```suggestion
    public static func equal(_ lhs: Angle, _ rhs: Angle, accuracy: Angle) -> Bool {
```

File: PathKit/Sources/ApproximatelyEquatable.swift [hunk 9]
```diff
@@ -0,0 +1,20 @@
...
+// contained in the file LICENSE at the root of the source code distribution
+// tree.
+
+import Foundation
+
+public protocol ApproximatelyEquatable {
+    associatedtype Accuracy
+    /// Returns a Boolean value indicating whether two values are equal within a given accuracy.
+    ///
+    ///
```
Comment (alejandro-isaza) [inline PathKit/Sources/ApproximatelyEquatable.swift]: remove extra empty line.

File: PathKit/Sources/ApproximatelyEquatable.swift [hunk 10]
```diff
@@ -0,0 +1,20 @@
...
+    /// - Parameters:
+    ///   - lhs: A value to compare.
+    ///   - rhs: Another value to compare.
+    ///   - accuracy: The accuracy used to determine if `lhs` is approximately equal to `rhs`.
+    static func equal(lhs: Self, rhs: Self, accuracy: Accuracy) -> Bool
```
Comment (alejandro-isaza) [inline PathKit/Sources/ApproximatelyEquatable.swift]:
```suggestion
    static func equal(_ lhs: Self, _ rhs: Self, accuracy: Accuracy) -> Bool
```

**Review by alejandro-isaza — CHANGES_REQUESTED** [review]

File: PathKit/Sources/Angle.swift (hunk 1 above)
Comment (jbeenie) [reply PathKit/Sources/Angle.swift]: will rebase

File: PathKit/Sources/Angle.swift (hunk 2 above)
Comment (jbeenie) [reply PathKit/Sources/Angle.swift]: done

File: PathKit/Sources/Angle.swift (hunk 1 above)
Comment (jbeenie) [reply PathKit/Sources/Angle.swift]: done

File: PathKit/Sources/Angle.swift (hunk 3 above)
Comment (jbeenie) [reply PathKit/Sources/Angle.swift]: done

File: PathKit/Sources/Angle.swift (hunk 5 above)
Comment (jbeenie) [reply PathKit/Sources/Angle.swift]: done

File: PathKit/Sources/Angle.swift (hunk 6 above)
Comment (jbeenie) [reply PathKit/Sources/Angle.swift]: done

File: PathKit/Sources/Angle.swift [hunk 11]
```diff
@@ -103,3 +125,15 @@ public func sin(_ angle: Angle) -> Double {
 public func tan(_ angle: Angle) -> Double {
     return tan(angle.radians)
 }
+
+extension Angle: Comparable {
+    public static func < (lhs: Angle, rhs: Angle) -> Bool {
+        return lhs.radians < rhs.radians
```
Comment (alejandro-isaza) [inline PathKit/Sources/Angle.swift]: I think this should be normalized:
```suggestion
        return lhs.radians. positiveNormalized < rhs.radians. positiveNormalized
```

**Review by alejandro-isaza — APPROVED** [review]

**Review by alejandro-isaza — APPROVED** [review]

File: PathKit/Sources/Angle.swift [hunk 12]
```diff
@@ -14,6 +14,9 @@ public struct Angle {
 
     public static let zero = Angle(radians: 0)
 
+    /// This value compares less than or equal to all Angles, but greater than Angle.zero.
+    public static let smallestAngle = Angle(radians: Double.leastNonzeroMagnitude)
```
Comment (alejandro-isaza) [inline PathKit/Sources/Angle.swift:18]: What about `leastNonzero` to conform to what Double has?

**Review by alejandro-isaza — APPROVED** [review]

**Pull request merged**